### DISCUSSION TITLE
Improve error message for incorrect arguments in defineStore

### DISCUSSION THREAD

**Discussion by bencodezen** [vuejs/pinia#855] [Ideas]
## 🧭  Context

So, I accidentally created a store without a storeId:

```js
export const useCounterStore = defineStore({ .... })
```

However, I got the following error in the console with things working kind of funny.

<img width="533" alt="Screen Shot 2021-11-29 at 8 54 30 AM" src="https://user-images.githubusercontent.com/4836334/143880468-52efb406-74c3-4b20-969a-87867e9445b1.png">

The solution ended up being a lack of an storeId, but the error doesn't really tell you this. In fact, I had no idea where to really start to debug it. However, luckily I realized the user error before reporting it as a bug. That said, I think we could improve this experience.

## 💡 Suggestion

1. Improve the error message.

If there is no store ID that's passed, then the error should throw an error that no ID was defined.

2. Generate an automatic store id with a warning that an automated has been generated and they should define their own.

If a user forgets to pass an ID, maybe we automatically generate a store with a hash like `pinia-store-2398hf`? This way it doesn't crash the app, but still warns the developer that they should define one.

**Comment by posva** [comment]
I'm not completely sure we should have a warning for this because in the end, it's the type of errors you catch at the typing level. So, the question would be: how easy is it to miss? If it does happen often, then, we could add it as a dev only feature with the automatic id generation. WDYT?

**Reply by bencodezen** [reply]
That makes sense! Is it possible for us to improve the error message on it to clarify what the problem is?

**Reply by posva** [reply]
BTW This might also be a reason to deprecate the syntax with only one argument (it's not used anymore in docs).

I haven't found other occurrences of the error in the discussions or issues, so I think it's too rare to add a specific warning for the moment

**Reply by bencodezen** [reply]
Sounds good!
